### PR TITLE
Merge satosa classes from ops repos

### PR DESCRIPTION
**Pull request by theseal**
Merge satosa classes from "all" ops repos so that we only need to maintain one class.

Can't find any ops repo that uses this class (`sunet::satosa`) so it should be safe to not be backwards compatible.

Some tweaks may be needed `cdb.json` when the class is used against a real Open Id Connect RP.

### CODE REVIEW

File: manifests/satosa.pp [hunk 1]
```diff
@@ -1,79 +1,121 @@
...
+      cert_file => '/etc/satosa/https.crt'
+    }
+  }
+
+  if ($enable_oidc) {
+    $client_id     = lookup('client_id')
+    $client_secret = lookup('client_secret')
+    file { '/etc/satosa/cdb.json':
+      ensure  => file,
+      content => template('sunet/satosa/cdb.json.erb')
```
Comment (mickenordin) [inline manifests/satosa.pp:102]: Should we make it possible to add your own cdb.json here? Possibly by adding a "manage_cdb" parameter to the class?

**Review by mickenordin — CHANGES_REQUESTED** [review]
Maybe my comment about being able to supply your own cdb.json should be a thing?

File: manifests/satosa.pp (hunk 1 above)
Comment (theseal) [reply manifests/satosa.pp:102]: I will try to completely rewrite this segment with you comment in mind. Stay tuned.

Comment (theseal) [reply manifests/satosa.pp]: A few commits later and the json configurations are now handled in the same way as all other yaml configuration in this class. Think it got a lot better.

Please have another look.

**Review by mickenordin — APPROVED** [review]
Looks good!

**Pull request merged**